src/main/java/com/google/step/servlets/JsonFormatter.java

```diff
@@ -214,4 +214,58 @@ public static String getUserJson(
   private static String getImageUrl(String blobKey) {
     return "/api/images/" + blobKey;
   }
+
+  /**
+   * Returns a json of home page data.
+   *
+   * @param homePageDealsMaps a list of list of maps for home page
+   * @return a json of home page data
+   */
+  public static String getHomePageJson(List<List<Map<String, Object>>> homePageDealsMaps) {
+    Gson gson = new Gson();
+    String json = gson.toJson(getHomePageMap(homePageDealsMaps));
+    return json;
+  }
+
+  /**
+   * Returns a map of sections for home page
+   *
+   * @param homePageDealsMaps a list of list of maps for home page
+   * @return a map of sections for home page
+   */
+  private static Map<String, Object> getHomePageMap(
+      List<List<Map<String, Object>>> homePageDealsMaps) {
+    Map<String, Object> homePageMap = new HashMap<>();
+    homePageMap.put("popularDeals", homePageDealsMaps.get(0));
+    homePageMap.put("usersIFollow", homePageDealsMaps.get(1));
+    homePageMap.put("restaurantsIFollow", homePageDealsMaps.get(2));
+    homePageMap.put("tagsIFollow", homePageDealsMaps.get(3));
```

Comment:
Remove this method and move this logic into the HomeServlet, this is too hardcoded